sdk/core/core-amqp/src/retry.ts

```diff
@@ -174,14 +204,30 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
         i,
         err
       );
+      let targetDelayInMs: number;
```

Comment:
We can initialize `targetDelayInMs` with `config.delayInSeconds` to avoid the else block